Algorithm.Framework/Alphas/PearsonCorrelationPairsTradingAlphaModel.cs

```diff
@@ -0,0 +1,99 @@
+        /// </summary>
+        /// <param name="lookback">Historical return lookback period</param>
+        /// <param name="resolution">The resolution of historical data</param>
+        /// <param name="threshold">The percent [0, 100] deviation of the ratio from the mean before emitting an insight</param>
+        public PearsonCorrelationPairsTradingAlphaModel(int lookback, Resolution resolution, decimal threshold = 1m)
+            : base(lookback, resolution, threshold)
+        {
+            _lookback = lookback;
+            _resolution = resolution;
+        }
+
+        /// <summary>
+        /// Event fired each time the we add/remove securities from the data feed
+        /// </summary>
+        /// <param name="algorithm">The algorithm instance that experienced the change in securities</param>
+        /// <param name="changes">The security additions and removals from the algorithm</param>
+        public override void OnSecuritiesChanged(QCAlgorithmFramework algorithm, SecurityChanges changes)
+        {
+            NotifiedSecurityChanges.UpdateCollection(Securities, changes);
+
+            var symbols = Securities.Select(x => x.Symbol).ToArray();
+
+            var history = algorithm.History(symbols, _lookback, _resolution)
+                .SelectMany(x => x.Bars.Values)
+                .GroupBy(x => x.Symbol)
+                .Select(x =>
+                {
+                    var array = x.Select(b => (double)b.Close).ToArray();
+
+                    for (var i = array.Length - 1; i > 0; i--)
+                    {
+                        array[i] = Math.Log(array[i]) - Math.Log(array[i - 1]);
+                    }
+                    array[0] = 0;
+
+                    return array;
+                });
+
+            var pearsonMatrix = Correlation.PearsonMatrix(history).UpperTriangle();
+
+            var corr = new Dictionary<Tuple<Symbol, Symbol>, double>();
+            var maxValue = pearsonMatrix.Enumerate().Where(x => Math.Abs(x) < 1).Max();
+            var maxTuple = pearsonMatrix.Find(x => x == maxValue);
+
+            _bestPair = Tuple.Create(symbols[maxTuple.Item1], symbols[maxTuple.Item2]);
+
+            base.OnSecuritiesChanged(algorithm, changes);
+        }
+
+        /// <summary>
+        /// Check whether the assets pass a pairs trading test
+        /// </summary>
+        /// <param name="algorithm">The algorithm instance that experienced the change in securities</param>
+        /// <param name="asset1">The first asset's symbol in the pair</param>
+        /// <param name="asset2">The second asset's symbol in the pair</param>
+        /// <returns>True if the statistical test for the pair is successful</returns>
+        public override bool HasPassedTest(QCAlgorithmFramework algorithm, Symbol asset1, Symbol asset2)
+        {
+            return asset1 == _bestPair.Item1 && asset2 == _bestPair.Item2;
```

Comment:
no change required -- I wonder if there's a 'threshold' kind of parameter missing here (or above when computing). In other words, perhaps we can have more than just the *best* pair, and instead select pairs that are above some threshold correlation coefficient.

Reply:
I will add such parameter. 
Ideally, we would look for a high Pearson correlation (over 0,95 perhaps), but the maximum found for those symbols were 0,75.